### PR TITLE
Frogtoberfest Completion Message Not Showing and Other Repo Invalid Count Still Being Counted

### PR DESCRIPTION
**Pull request by bhattaraib58**
## Issue
Although Frogtoberfest completion message showing based on condition was fixed recently, maybe due to some merge issue the message was not showing.

And again the invalid PR in other repos was still being counted.

## Usage Changes
For showing frogtoberfest completion message Binded setUserContributionCount to parent this.
For other repo invalid count still being counted, again validated the data before counted other repo count.

## Screenshot
Before:
Screenshot from 2019-10-26 21-59-03

After:
Screenshot from 2019-10-26 21-53-09

Screenshot from 2019-10-26 21-53-22

### CODE REVIEW

File: src/pages/User/index.js [hunk 1]
```diff
@@ -9,10 +9,21 @@ import PullRequests from './components/PullRequests';
  * User Component.
  */
 export class User extends Component {
-  state = {
```
Comment (kabirbaidhya) [inline src/pages/User/index.js]: Let's keep this. It's fine to initialize the `state` like this.

File: src/pages/User/index.js [hunk 2]
```diff
@@ -9,10 +9,21 @@ import PullRequests from './components/PullRequests';
...
+   *
+   * @param {*} props
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      totalPrCount: 0,
+      totalOtherPrCount: 0
+    };
```
Comment (kabirbaidhya) [inline src/pages/User/index.js]: Undo.

File: src/pages/User/index.js [hunk 3]
```diff
@@ -9,10 +9,21 @@ import PullRequests from './components/PullRequests';
...
+
+    this.setUserContributionCount = this.setUserContributionCount.bind(this);
```
Comment (kabirbaidhya) [inline src/pages/User/index.js]: `bind(this)` - I don't understand why this wasn't needed before and why we need it now - also how does it solve the issue?

File: src/pages/User/index.js (hunk 1 above)
Comment (bhattaraib58) [reply src/pages/User/index.js]: ok dai.

File: src/pages/User/components/PullRequests/index.js [hunk 4]
```diff
@@ -180,13 +180,16 @@ class PullRequests extends Component {
    */
   displayPullRequests = userInfo => {
     const { data, userDetail } = userInfo;
-    const count = this.countOtherRepos(data, userDetail);
+    const validData = this.getValidPullRequests(data);
+    const otherReposCount = this.countOtherRepos(validData, userDetail);
+
+    this.props.setUserContributionCount(validData.items.length, otherReposCount);
```
Comment (kabirbaidhya) [inline src/pages/User/components/PullRequests/index.js]: I doubt this line is still needed? Why? 
I think the ordering of the above lines and the set state below should fix this? Can you check and verify if this is the case?

File: src/pages/User/index.js (hunk 3 above)
Comment (bhattaraib58) [reply src/pages/User/index.js]: As Component user had two child Pullrequests and UsernameForm.

setUserContributionCount was parent method it was sent to child component Pullrequests which has set the totalOtherprcount and totalPrCount values, 
the data which later passed on to the UsernameForm where the completion message was shown. As being different component parent was used to communicate between child.

And without "this"  binding the function passed to child only set value to itself not on parent.

File: src/pages/User/components/PullRequests/index.js (hunk 4 above)
Comment (bhattaraib58) [reply src/pages/User/components/PullRequests/index.js]: Hmm, sorry i didn't understand , can you tell which lines should be reordered and set state below line ??

Comment (kabirbaidhya) [reply src/pages/User/components/PullRequests/index.js]: I mean - I just didn't get why line 186 was needed. All other lines look OK to me. 

Can you remove this line :point_down: and check if you can still make it work? 
```
this.props.setUserContributionCount(validData.items.length, otherReposCount);
```

Comment (bhattaraib58) [reply src/pages/User/components/PullRequests/index.js]: No Dai ! Without setting value in setUserContributionCount, we can't set contribution count for checking the eligibility criteria for showing CompletionMessage or Time message in UsernameForm component . For which on UsernameForm component default state value will be passed 0,0 for totalprCount and otherprCount which will always give display timeMessage.

Screenshot from 2019-10-26 22-48-38

**Pull request merged**